span.go

```diff
@@ -113,7 +116,14 @@ func (s *spanImpl) Log(ld opentracing.LogData) {
 		ld.Timestamp = time.Now()
 	}
 
-	s.raw.Logs = append(s.raw.Logs, ld)
+	if maxLogs := s.tracer.options.MaxLogsPerSpan; maxLogs > 0 && len(s.raw.Logs) == maxLogs {
+		// We have too many logs. Overwrite the earliest log (use Logs as a circular
+		// buffer).
+		s.raw.Logs[s.numDroppedLogs%maxLogs] = ld
```

Comment:
Interesting. Would that be about half of max before the dropped ones, and half after?
It's easy enough to implement whichever policy on my end; I will wait until there is agreement. It might be easier to access the discussion post-mortem if it's in #38 instead of here.